node/libs/protobuf_build/src/lib.rs

```diff
@@ -52,70 +56,68 @@ fn traverse_files(
     Ok(())
 }
 
-/// Protobuf descriptor + info about the mapping to rust code.
+/// Manifest of a Protobuf compilation target containing information about the compilation process.
 #[derive(Debug)]
-pub struct Descriptor {
+struct Manifest {
     /// Root proto package that all proto files in this descriptor belong to.
-    /// Rust types have been generated relative to this root.
-    proto_root: ProtoName,
-    /// Raw descriptor proto.
-    descriptor_proto: prost_types::FileDescriptorSet,
-    /// Direct dependencies of this descriptor.
-    dependencies: Vec<&'static Descriptor>,
+    proto_root: ProtoPath,
+    /// Absolute path to the descriptor.
+    descriptor_path: String,
+    /// Tuples of `proto_root` and absolute paths to the corresponding descriptor for all dependencies
+    /// including transitive ones.
+    dependencies: Vec<(ProtoPath, String)>,
 }
 
-impl Descriptor {
-    /// Constructs a Descriptor.
-    pub fn new(
-        proto_root: ProtoName,
-        dependencies: Vec<&'static Descriptor>,
-        descriptor_bytes: &[u8],
-    ) -> Self {
-        Descriptor {
-            proto_root,
-            dependencies,
-            descriptor_proto: prost_types::FileDescriptorSet::decode(descriptor_bytes).unwrap(),
-        }
-    }
+impl Manifest {
+    /// Loads manifest from the environment variable.
+    fn from_env(rust_root: &RustName) -> anyhow::Result<Self> {
+        let crate_name = rust_root.crate_name().context("empty `rust_root`")?;
+        let env_name = format!("DEP_{}_PROTO_MANIFEST", crate_name.to_uppercase());
```

Comment:
wouldn't serde serialization work better than custom serialization here?

Reply:
I've used the `serde` serialization initially, but decided against it since the stuff we're (de)serializing is essentially a sequence of `(ProtoPath, Path)` tuples (i.e., has a very simple structure).

Reply:
ack